### DISCUSSION TITLE
Please help me to understand how to tune VT and TRVs

### DISCUSSION THREAD

**Discussion by pieri70** [jmcollin78/versatile_thermostat#326] [Q&A]
Hello everyone
I'm new to home automation, Home Assistant, and Versatile Thermostat, so please be patient ;)
I come from Better thermostat that made me crazy trying to setup my heating system in Home Assistant.

I want to control heating on 4 rooms, 3 with a single radiator and one with 2 radiators, 5 Sonoff TRVZB TRVs (zigbee).
One automation controls if any of these 5 TRV is in heating state and turns on the boiler.
The boiler is turned off if all TRVs are in idle state.
I made 3 VT istances with just underlying_climate_0 and one with both underlying_climate_0 and underlying_climate_1 (2 radiators/TRV on a single room).

My system works but there is some mess in setting the TRV target temperature and offset.

These TRV have "Local temperature calibration, i.e. Offset to add/subtract to the local temperature" that is a value I can change to let the local temperaturea of a given TRV to be near the real room themperature, in my case the temperature from a second temperature/humidity sensor (the one used in VT as "temperature sensor entity identifier which gives the temperature of the room in which the radiator is installed" - as for VT instructions).

As indicated by jmcollin78 in another discussion I set strong Self-regulation on my VTs because I had some problems controlling the state of the underlying TRVs.
I continue to have some issues because I set values to "Local temperature calibration" for my TRVs, trying to make their local temperature to be near to the "real" room temperature and I see VT that don't turn ON/OFF underlaying TRV when needed.
I think I'm making some mistake..

So, could you please explain me how does VT controls the temperature on underlaying TRVs?

Does it change only the target temperature on the TRV or does it also change the "Local temperature calibration, i.e. Offset to add/subtract to the local temperature" (this last doesn't seem to me, because values remain as I set them manually in zigbee2mqtt)

Is it correct to manually set Local temperature calibration on each TRV?
If yes, how do you adjust it?
Do you adjust it after a long idle period, so the radiator is not hot and the temperature is "uniform" in the room?
Do you adjust it after a heating cycle?
Do you adjust it during heating?

Do you set Local temperature calibration = 0 (no local calibration/offset) also if the temperature detected by TRV internal thermometer is 2-3°C different (usually lower) from room sensor?

Any help is appreciated
Thanks

**Comment by pieri70** [comment]
For example now I am in this situation in kitchen

![image](https://github.com/jmcollin78/versatile_thermostat/assets/11612625/bdbb374b-0bf7-4afd-b80d-99b21d469750)


Heating is on because the TRV wants it and VT isn't requesting it....

I set -3°C offset for that TRV because this morning that was the value to make room sensor temperature similar to TRV sensor temperature..

![image](https://github.com/jmcollin78/versatile_thermostat/assets/11612625/64dd2154-01c0-4e7e-ad5a-e28f0971dbf1)

I'm going nuts..

**Comment by jmcollin78** [comment]
Hello @pieri70 ,

I will try to help you.

> Does it change only the target temperature on the TRV or does it also change the "Local temperature calibration

VTherm only send the target temp to the TRV. It don't change the calibration and don't start or stop automatically the underlying TRV but follow the underlying state. If you turns off the TRV on the TRV directly this will turn off the VTherm.

> Is it correct to manually set Local temperature calibration on each TRV?

No. For me you don't need to use that. Just let the Strong regulation do its job. The Strong regulation will adjust in real an offset calculated depending on conditions.
So for me, remove the Local temperature calibration (a static value cannot be good as conditions vary over time), and just configure the self-regulation into Strong mode.

Have a thermostat cycle which is no more than 10 min.
Configure self-regulation with those values (to avoid drowing your TRV battery) :
![Capture d’écran 2024-01-09 à 08 27 15](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/7a682304-756b-4001-87ee-dd027d082982)

Try this and implement some graphs as explained in the README (see Plotly). Post the graph you have after stabilisation and we will see if there is something else to change.

**Comment by pieri70** [comment]
Super thanks.
I will try.
I guess VT takes some time to calculate correct PID for a given VRT, right?
So the effects of the calibration should be seen after some time (days?)

**Reply by jmcollin78** [reply]
Yes it take some times, but it should be in one or two hours depending of the latency of your heaters.

**Comment by pieri70** [comment]
Post cancelled I had some problems with my boiler and one TRV defective
Stay tuned.
Thanks

**Comment by pieri70** [comment]
Hello
I'm back with situation that has set a little and I can control all trvs.
Now ?im facing a problem,
Here is a history graph, 4 VT controlling TRVS and the bottom one is the boiler on/off state (controlled by an automation based on heating/idle of single TRVs)
 
![image](https://github.com/jmcollin78/versatile_thermostat/assets/11612625/b1f9562c-83a2-4328-9e2e-88a933e4acd0)

And this is a graph showing on/off time for the 4 VT (bars) and the main boiler control (line)

![image](https://github.com/jmcollin78/versatile_thermostat/assets/11612625/3d513ab6-1160-4af2-9708-80b4ee964fb4)

The boiler is the result union of the single trvs heating/idle sates.
As some user on my italian home assistant forum suggested me it would be better if the boiler will stay on for longer time at lower water temperature sent to radiators.
This way the boiler should modulate its power and should consume less gas..

I am planning to lower the water temperature setup on my boiler but I would like to know if there is some parameters on VT that can make heating periods longer.
Thanks

**Reply by jmcollin78** [reply]
Hello @pieri70 ,

It depends of the type of VTherm you have.

In `over_valve`:
you have only the cycle parameters to help you:
1. the cycle in minute: at each cycle, the conditions are examined, the % opening the valve is recalculated,

In `over_switch`, you have 2 parameters:
1. the cycle as above,
2. and the most important for you case: 
![Capture d’écran 2024-01-16 à 08 40 28](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/886043e9-90c3-4311-b68b-844158114702)

This controls the minimal delay of turning on a heater (useful for electrical heater),

In `over_climate` with self-regulation turned on:
1. the self-regulation threshold (don't change anything for small changes),
2. the self-regulation frequency : don't change too often.

See documentation here: https://github.com/jmcollin78/versatile_thermostat/blob/feature_234-add_central_boiler_helper/README.md#self-regulation

Without self-regulation, there is no parameter to control anything because the regulation is done by the underlying itself (so you should tune it directly if possible).

**Reply by pieri70** [reply]
Thanks @jmcollin78 
I am in over_climate with strong regulation.

Here is one of my vtherm which has many cycles/day
at 7 my wife opens the window for 15 minutes. Then temperature takes a lot to rise and aftef 14 (comfort) the rise is even slower with VT that don't remain on for long time..

![image](https://github.com/jmcollin78/versatile_thermostat/assets/11612625/eb656167-81ed-4c92-8dca-98f47e83f145)


```
hvac_modes: off, auto, heat
min_temp: 8
max_temp: 28
target_temp_step: 0.5
preset_modes: none, frost, eco, comfort, boost
current_temperature: 17.6
temperature: 18
hvac_action: heating
preset_mode: comfort
is_on: true
hvac_mode: heat
type: null
is_controlled_by_central_mode: false
last_central_mode: null
frost_temp: 7
eco_temp: 17
boost_temp: 19
comfort_temp: 18
frost_away_temp: 12
eco_away_temp: 12
boost_away_temp: 12
comfort_away_temp: 12
power_temp: null
target_temperature_step: 0.5
ext_current_temperature: 4.84
ac_mode: false
current_power: null
current_power_max: null
saved_preset_mode: comfort
saved_target_temp: 18
saved_hvac_mode: heat
window_state: off
motion_state: null
overpowering_state: null
presence_state: null
window_auto_state: off
window_bypass_state: false
security_delay_min: 60
security_min_on_percent: 0.5
security_default_on_percent: 0.1
last_temperature_datetime: 2024-01-16T18:40:52.867741+01:00
last_ext_temperature_datetime: 2024-01-16T18:34:13.503312+01:00
security_state: false
minimal_activation_delay_sec: 10
device_power: 1
mean_cycle_power: null
total_energy: 11.49
last_update_datetime: 2024-01-16T18:51:31.278191+01:00
timezone: Europe/Rome
window_sensor_entity_id: binary_sensor.finestra_davide_contact
window_delay_sec: 30
window_auto_enabled: false
window_auto_open_threshold: 3
window_auto_close_threshold: 0
window_auto_max_duration: 30
motion_sensor_entity_id: null
presence_sensor_entity_id: null
power_sensor_entity_id: null
max_power_sensor_entity_id: null
temperature_unit: °C
is_device_active: true
ema_temp: 17.53
is_over_climate: true
start_hvac_action_date: 2024-01-16T18:51:31.272256+01:00
underlying_climate_0: climate.trv_davide
underlying_climate_1: null
underlying_climate_2: null
underlying_climate_3: null
is_regulated: true
regulated_target_temperature: 22
auto_regulation_mode: auto_regulation_strong
regulation_accumulated_error: 50
auto_fan_mode: auto_fan_none
current_auto_fan_mode: auto_fan_none
auto_activated_fan_mode: null
auto_deactivated_fan_mode: null
friendly_name: Camera Davide
supported_features: 17
```

**Comment by pieri70** [comment]
So, let me see if I understand..
I'm in over_climate so:
First setup mask, cycle duration (minutes) don't do anything, it works for over_valve
![image](https://github.com/jmcollin78/versatile_thermostat/assets/11612625/1a47ca44-a210-431c-a87a-f9da8608b536)

On underlying trv there is auto regulation with its parameters but as I understand correctly they don't do a lot on on/off periods, right?

![image](https://github.com/jmcollin78/versatile_thermostat/assets/11612625/e58ef805-9096-453f-ac64-54be69b550b1)

Under advanced parameters I only see safety parameters.
Should I work on my boiler only? Lower water temperature so it takes longer to worm rooms?

**Reply by jmcollin78** [reply]
> On underlying trv there is auto regulation with its parameters but as I understand correctly they don't do a lot on on/off periods, right?

If you change this parameters, it will change the number of changes on target temp change on the TRV. It is more to avoid drowning the battery of the TRV when regulation is done too often.

It doesn't change anything on the cycle itself, because the regulation is done by the underlying climate which controls the TRV. But it can change the frequency of the changes.

So as said above, in `over_climate` there is no more tuning parameters.

> Should I work on my boiler only? Lower water temperature so it takes longer to worm rooms?

I have no idea. I don't have any central boiler nor experience returns on this point.  If you have an answer, I am interested of course.

**Reply by pieri70** [reply]
So it would be better to use a TRV with opening control (over_valve)?
For now I tried to lower the boiler temperature, I will see if in the next days trv will react differently
Thanks

**Reply by jmcollin78** [reply]
> So it would be better to use a TRV with opening control (over_valve)?

For me it is better because we have a full control of the regulation. We can also control the boiler to start it when one of the TRV is open. But it is expensive and you have to change all your TRV for this. With over_valve you will also have the cycle parameter control.
